debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/AbstractReader.java

```diff
@@ -44,12 +46,13 @@
     protected final MySqlJdbcContext connectionContext;
     private final BlockingQueue<SourceRecord> records;
     private final AtomicBoolean running = new AtomicBoolean(false);
+    private final AtomicReference<MySqlPartition> partition = new AtomicReference<>();
```

Comment:
Is it strictly needed to touch the legacy reader code? We're planning to phase it out as of Debezium 2.0 anyways.

Reply:
I would love not to touch it but it seems inevitable. For instance, consider the following code execution flow in reverse order:

1. https://github.com/debezium/debezium/blob/1484980bf906b97305bdc542b84b304fed87e2ff/debezium-core/src/main/java/io/debezium/pipeline/metrics/PipelineMetrics.java#L64
2. https://github.com/debezium/debezium/blob/1aa7b7b426864dce077907fe3a006e3236dde023/debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/BinlogReader.java#L677

The framework-level `PipelineMetrics` now require a partition, so the legacy `BinlogReader` has to pass it somehow.

Of course, instead of passing the partition through the entire flow in the legacy component, we could pass some dummy value where the legacy code directly integrates with the core APIs but it would be much more error-prone and hard to maintain even in the short term.

Reply:
Yeah, makes sense.